knative/networking
# proxy protocol probe mvp

# Changes



- Adds proxy protocol support when probing


/kind enhancement


Fixes
https://github.com/knative/serving/issues/13123
https://github.com/knative-sandbox/net-istio/issues/463



This is my first contribution to networking directly and I have a fair amount of questions, so I'll try and add tasks for questions that I think should be answered before this gets merged.



**New Dependency**

I've added a new [third party dependency](https://github.com/pires/go-proxyproto) that handles the proxy protocol header construction.

I've added the license file to the third party folder, the project is licensed under Apache 2.0 so I assume no issues here.

**Initial testing**

Initial testing scenario is as follows:

1. Knative serving installed normally via yaml
2. net-istio picking up this set of changes locally, use ko to apply the new net-istio controller to my cluster
3. Configure Magic DNS / sslip
4. [apply the proxy protocol envoy filter as seen in the istio documentation](https://istio.io/latest/docs/ops/configuration/traffic-management/network-topologies/#proxy-protocol)
5. create a new test service
6. observe that the test service becomes ready, and responds to normal network traffic

**Design Discussion**

Refactored this to not support the offer/doAsync portion - they don't need to be supported as it's only used by serving's PA and that probes via private service.

Let me know if there's a better way to consume the `config-network` values.
Needed to add a volume mount to support the current approach https://github.com/knative-sandbox/net-istio/pull/1028

Will need to test with Contour but I imagine it's the same as Istio - just need to consume the configmap.
